test/run_test.py

```diff
@@ -1273,7 +1265,11 @@ def parse_args():
         help="Run tests with TorchInductor turned on",
     )
 
-    return parser.parse_args()
+    args, extra = parser.parse_known_args()
+    if "--" in extra:
+        extra.remove("--")
```

Comment:
When might '--' be specified? A test class/name filter is one example. Are there others that might end up broken by this change?

Reply:
I don't think `--` is actually required for a test class filter, afaik you can just `python file.py Test.test` and it'll work.  I was under the impression that `--` was syntax for "the args after this should be given to the subprocess" but argparse will include things before it if they're not known so I'm honestly not quite sure what convention is

Reply:
I was thinking about this too especially as for the removal of the non-positional argument which has this example:  
`python run_test.py -i sparse -- TestSparse.test_factory_size_check`

Given your explanation that will likely continue to work. Well not only that: It will fail without it: We append later to this list so without this we'll get `-- TestSparse.test_factory_size_check --shard-id=1` which will break the logic